xbmc/addons/addoninfo/AddonType.h

```diff
@@ -18,74 +18,76 @@ class TiXmlElement;
 namespace ADDON
 {
 
-typedef enum
+enum class AddonType
 {
-  ADDON_UNKNOWN,
-  ADDON_VIZ,
-  ADDON_SKIN,
-  ADDON_PVRDLL,
-  ADDON_INPUTSTREAM,
-  ADDON_GAMEDLL,
-  ADDON_PERIPHERALDLL,
-  ADDON_SCRIPT,
-  ADDON_SCRIPT_WEATHER,
-  ADDON_SUBTITLE_MODULE,
-  ADDON_SCRIPT_LYRICS,
-  ADDON_SCRAPER_ALBUMS,
-  ADDON_SCRAPER_ARTISTS,
-  ADDON_SCRAPER_MOVIES,
-  ADDON_SCRAPER_MUSICVIDEOS,
-  ADDON_SCRAPER_TVSHOWS,
-  ADDON_SCREENSAVER,
-  ADDON_PLUGIN,
-  ADDON_REPOSITORY,
-  ADDON_WEB_INTERFACE,
-  ADDON_SERVICE,
-  ADDON_AUDIOENCODER,
-  ADDON_CONTEXT_ITEM,
-  ADDON_AUDIODECODER,
-  ADDON_RESOURCE_IMAGES,
-  ADDON_RESOURCE_LANGUAGE,
-  ADDON_RESOURCE_UISOUNDS,
-  ADDON_RESOURCE_GAMES,
-  ADDON_RESOURCE_FONT,
-  ADDON_VFS,
-  ADDON_IMAGEDECODER,
-  ADDON_SCRAPER_LIBRARY,
-  ADDON_SCRIPT_LIBRARY,
-  ADDON_SCRIPT_MODULE,
-  ADDON_GAME_CONTROLLER,
-  ADDON_VIDEOCODEC,
+  ADDON_UNKNOWN = 0,
+  ADDON_VIZ = 1,
+  ADDON_SKIN = 2,
+  ADDON_PVRDLL = 3,
+  ADDON_INPUTSTREAM = 4,
+  ADDON_GAMEDLL = 5,
+  ADDON_PERIPHERALDLL = 6,
+  ADDON_SCRIPT = 7,
+  ADDON_SCRIPT_WEATHER = 8,
+  ADDON_SUBTITLE_MODULE = 9,
+  ADDON_SCRIPT_LYRICS = 10,
+  ADDON_SCRAPER_ALBUMS = 11,
+  ADDON_SCRAPER_ARTISTS = 12,
+  ADDON_SCRAPER_MOVIES = 13,
+  ADDON_SCRAPER_MUSICVIDEOS = 14,
+  ADDON_SCRAPER_TVSHOWS = 15,
+  ADDON_SCREENSAVER = 16,
+  ADDON_PLUGIN = 17,
+  ADDON_REPOSITORY = 18,
+  ADDON_WEB_INTERFACE = 19,
+  ADDON_SERVICE = 20,
+  ADDON_AUDIOENCODER = 21,
+  ADDON_CONTEXT_ITEM = 22,
+  ADDON_AUDIODECODER = 23,
+  ADDON_RESOURCE_IMAGES = 24,
+  ADDON_RESOURCE_LANGUAGE = 25,
+  ADDON_RESOURCE_UISOUNDS = 26,
+  ADDON_RESOURCE_GAMES = 27,
+  ADDON_RESOURCE_FONT = 28,
+  ADDON_VFS = 29,
+  ADDON_IMAGEDECODER = 30,
+  ADDON_SCRAPER_LIBRARY = 31,
+  ADDON_SCRIPT_LIBRARY = 32,
+  ADDON_SCRIPT_MODULE = 33,
+  ADDON_GAME_CONTROLLER = 34,
+  ADDON_VIDEOCODEC = 35,
 
   /**
     * @brief virtual addon types
     */
   //@{
-  ADDON_VIDEO,
-  ADDON_AUDIO,
-  ADDON_IMAGE,
-  ADDON_EXECUTABLE,
-  ADDON_GAME,
+  ADDON_VIDEO = 36,
```

Comment:
From overview looks your change about request nice and good code improvements.

Only about the numbering of the enums I'm a bit unsure. There it becomes more hard to have numbering correct if another type becomes added, as the virtual types always on end. Think better to leave as before and let the compiler to make the number.

Also as it becomes added to an enum class, maybe to make it shorter? E.g. as `AddonType::UNKNOWN`, `AddonType::PVR`, `AddonType::VIDEO`..., think the `ADDON_...` then no more needed.

Reply:
> Only about the numbering of the enums I'm a bit unsure.

I will remove the numbers. They are not needed.

> Also as it becomes added to an enum class, maybe to make it shorter?

Okay. I will remove the ADDON_ prefix and fix potential compile errors (caused by some dumb defines like MAX on Windows).

Reply:
done (separate commit)